### ISSUE TITLE
Add `nn::AdaptiveAvgPool2D`

### ISSUE THREAD

**Issue by coreylowman** [coreylowman/dfdx#215]
See https://pytorch.org/docs/stable/generated/torch.nn.AdaptiveAvgPool2d.html

This takes fixed output size and determines kernel size based on input. This should be possible with const generics & using avg pool from #157

**Comment by coreylowman** [comment]
See pytorch impl here: https://github.com/pytorch/pytorch/blob/51861cc9b19d9c483598e39932661822a826d3a2/aten/src/ATen/native/AdaptiveAveragePooling.cpp#L21.

See https://stackoverflow.com/questions/53841509/how-does-adaptive-pooling-in-pytorch-work/63603993#63603993 for info about non-integer strides:
- Stride = (input_size//output_size)
- Kernel size = input_size - (output_size-1)*stride
- Padding = 0

**Comment by coreylowman** [comment]
Going to close this for now. Most things use the equivalent of GlobalAvgPool, which is already added.

If someone needs this, feel free to reopen

**Comment by quietlychris** [comment]
I was working on implementing AlexNet in `dfdx`per https://github.com/LaurentMazare/tch-rs/blob/main/src/vision/alexnet.rs, but have gotten a bit blocked by not having this layer type available. Would it be possible to add this, or alternatively to have some documentation on constructing the equivalent using `GlobalAvgPool` that you mentioned above?
